tests/validate-urls.rb

```diff
@@ -3,20 +3,25 @@
 
 require 'English'
 require 'json'
-@status = 0
+require 'httpclient'
+status = 0
 
 # Fetch created/modified files in entries/**
 diff = `git diff --name-only --diff-filter=AM origin/master...HEAD entries/`.split("\n")
 
 def curl(url)
-  `curl --fail -A "Mozilla/5.0 (compatible;  MSIE 7.01; Windows NT 5.0)" -H "FROM: https://2fa.directory" #{url}`
-  # rubocop:disable Style/GuardClause
-  unless $CHILD_STATUS.success?
-    # Break build if above cURL exited with non-zero value
-    puts "::error file=#{@path}:: Unable to reach #{url}"
-    @status = 1
-  end
-  # rubocop:enable Style/GuardClause
+  headers = { 'User-Agent' => 'Mozilla/5.0 (compatible;  MSIE 7.01; Windows NT 5.0)', 'FROM' => '2fa.directory' }
+  req = HTTPClient.new
+  req.receive_timeout = 8
+  return 0 if (res = req.get(url, nil, headers, follow_redirect: true).status == 200)
+
+  raise(nil) unless res.status.match?('/50\d/')
```

Comment:
```suggestion
  raise(nil) unless res.status.to_s.match(/50\d|403/)
```

The pattern wasn't being detected due to the quotes and there was an error with trying to match an integer. Also added 403 codes to be warned for, as those were also common when bot detection scripts were blocking (see #6123)